docs/articles_en/documentation/openvino-ir-format/operation-sets/pooling_shape_rules.rst

```diff
@@ -0,0 +1,208 @@
+
+
+2. Example 2 shows how *MaxPool* operates with 3D input using 1D kernel and ``auto_pad = valid``.
+
+   .. code-block:: sh
+
+      input = [[[-1, 2, 3, 5, -7, 9, 1]]]   # shape: (1, 1, 7)
+      strides = [1]
+      kernel = [3]
+      rounding_type = "floor"
+      auto_pad = "valid"
+      output0 = [[[3, 5, 5, 9, 9]]]   # shape: (1, 1, 5)
+      output1 = [[[2, 3, 3, 5, 5]]]   # shape: (1, 1, 5)
+
+
+3. Example 3 shows how *MaxPool* operates with 4D input using 2D kernel and ``auto_pad = same_lower``.
+
+   .. code-block:: sh
+
+      input = [[[[-1, 2, 3],
+               [4, 5, -6],
+               [-7, 8, 9]]]]   # shape: (1, 1, 3, 3)
+      strides = [1, 1]
+      kernel = [2, 2]
+      rounding_type = "floor"
+      auto_pad = "same_lower"
+      output0 = [[[[-1, 2, 3],
+                  [4, 5, 5]
+                  [4, 8, 9]]]]   # shape: (1, 1, 3, 3)
+      output1 = [[[[0, 1, 2],
+                  [3, 4, 4],
+                  [3, 7, 8]]]]   # shape: (1, 1, 3, 3)
+
+
+4. Example 4 shows how *MaxPool* operates with 4D input using 2D kernel and ``auto_pad = same_upper``.
+
+   .. code-block:: sh
+
+      input = [[[[-1, 2, 3],
+                 [4, 5, -6],
+                 [-7, 8, 9]],
+                [[2, -1, 5],
+                 [6, -7, 1],
+                 [8, 2, -3]]]]   # shape: (1, 2, 3, 3)
+      strides = [1, 1]
+      kernel = [2, 2]
+      rounding_type = "floor"
+      auto_pad = "same_upper"
+      output0 = [[[[5, 5, 3],
+                   [8, 9, 9]
+                   [8, 9, 9]],
+                  [[6, 5, 5],
+                   [8, 2, 1],
+                   [8, 2, -3]]]]   # shape: (1, 2, 3, 3)
+      output1 = [[[[4, 4, 2],
+                   [7, 8, 8],
+                   [7, 8, 8]],
+                  [[12, 11, 11],
+                   [15, 16, 14],
+                   [15, 16, 17]]]]   # shape: (1, 2, 3, 3)
+
+
+5. Example 5 shows how *MaxPool* operates with 4D input using 2D kernel and ``rounding_type = ceil_torch``.
+
+   .. code-block:: sh
+
+      input = [[[[1, 2, 3],
+                 [4, 5, 6],
+                 [7, 8, 9]]]]   # shape: (1, 1, 3, 3)
+      strides = [2, 2]
+      kernel = [2, 2]
+      pads_begin = [1, 1]
+      pads_end = [1, 1]
+      rounding_type = "ceil_torch"
+      output0 = [[[[1, 3],
+                   [7, 9]]]]   # shape: (1, 2, 2, 2)
+      output1 = [[[[0, 2],
+                   [6, 8]]]]   # shape: (1, 2, 2, 2)
+
+
+6. Example 6 shows how *MaxPool* operates with 4D input using 2D kernel, ``auto_pad = valid`` and ``rounding_type = ceil``.
+
+   .. code-block:: sh
+
+      input = [[[[-1, 2, 3],
+                 [4, 5, -6],
+                 [-7, 8, 9]]]]   # shape: (1, 1, 3, 3)
+      strides = [2, 2]
+      kernel = [2, 2]
+      rounding_type = "ceil"
+      auto_pad = "valid"
+      output0 = [[[[5, 3],
+                   [8, 9]]]]   # shape: (1, 2, 2, 2)
+      output1 = [[[[4, 2],
+                   [7, 8]]]]   # shape: (1, 2, 2, 2)
```

Comment:
As above, there are 4 elements in the output, but the described shape has `1x2x2x2 = 8`
```suggestion
      output0 = [[[[5, 3],
                   [8, 9]]]]   # shape: (1, 1, 2, 2)
      output1 = [[[[4, 2],
                   [7, 8]]]]   # shape: (1, 1, 2, 2)
```